core/src/main/groovy/noe/tomcat/configure/SecureHttpConnectorTomcat.groovy

```diff
@@ -78,39 +116,82 @@ public class SecureHttpConnectorTomcat extends ConnectorTomcatAbstract<SecureHtt
     return this.sslPassword
   }
 
+  public String getSslImplementationName() {
+    return this.sslImplementationName
+  }
+
+  public Boolean getSslEnabledProtocols() {
+    return this.sslEnabledProtocols
+  }
+
   public SecureHttpConnectorTomcat setSslProtocol(String sslProtocol) {
     this.sslProtocol = sslProtocol
     return this
   }
 
-  public SecureHttpConnectorTomcat setKeystoreFile(String keystoreFile) {
+  public SecureHttpConnectorTomcat setKeystoreFile(String keystoreFile = keystoreFilePath) {
```

Comment:
Could you please do not use internal class properties as default method values? I understand that it is possible in Groovy but the code is a little bit foggy because of that. Do not hesitate to do default initialization in the method body.

Reply:
What do you have in mind? Because if I add default initialization in the method body, won't I risk ignoring the user-given variable ?Or did you mean default initialization in the construction body ?
Because my target was being able to call setKeystoreFile() without providing a String. If I remove the Groovy default initialization, this is not possible.

Reply:
Ok, I think I see your point better now. You want to provide some default values for certificates and keystores, right?
And you want to set the default in case that smbd call the setter w/t arguments.

To be honest, I don't prefer this way of implementation because it is not semantic and is not transparent - argumentless method calling sets something -and the method must be called explicitly to be an "unknown" default value set.

I would rather see smth more semantic and transparent.

Reply:
Thanos, you could add a new method called setDefaultKeyStoreFile() something like this:
`
public SecureHttpConnectorTomcat setDefaultKeyStoreFile() {
     setKeystoreFile(keystoreFilePath)
 }
`